packages/client/src/runtime/core/errorRendering/applyValidationError.ts

```diff
@@ -0,0 +1,494 @@
+      applyIncludeOnScalarError(error, args)
+      break
+    case 'EmptySelection':
+      applyEmptySelectionError(error, args)
+      break
+    case 'UnknownSelectionField':
+      applyUnknownSelectionFieldError(error, args)
+      break
+    case 'UnknownArgument':
+      applyUnknownArgumentError(error, args)
+      break
+    case 'UnknownInputField':
+      applyUnknownInputFieldError(error, args)
+      break
+    case 'RequiredArgumentMissing':
+      applyRequiredArgumentMissingError(error, args)
+      break
+    case 'InvalidArgumentType':
+      applyInvalidArgumentTypeError(error, args)
+      break
+    case 'InvalidArgumentValue':
+      applyInvalidArgumentValueError(error, args)
+      break
+    case 'SomeFieldsMissing':
+      applySomeFieldsMissingError(error, args)
+      break
+    case 'TooManyFieldsGiven':
+      applyTooManyFieldsGivenError(error, args)
+      break
+    case 'Union':
+      applyUnionError(error, args)
+      break
+    default:
+      console.log(error)
+      throw new Error('not implemented: ' + error.kind)
+  }
+}
+
+function applyIncludeAndSelectError(error: IncludeAndSelectError, argsTree: ArgumentsRenderingTree) {
+  const object = argsTree.arguments.getDeepSubSelectionValue(error.selectionPath)
+  if (object && object instanceof ObjectValue) {
+    object.getField('include')?.markAsError()
+    object.getField('select')?.markAsError()
+  }
+
+  argsTree.addErrorMessage(
+    (chalk) =>
+      `Please ${chalk.bold('either')} use ${chalk.greenBright('`include`')} or ${chalk.greenBright(
+        '`select`',
+      )}, but ${chalk.redBright('not both')} at the same time.`,
+  )
+}
+
+function applyIncludeOnScalarError(error: IncludeOnScalarError, argsTree: ArgumentsRenderingTree) {
+  const [selectionPath, field] = splitPath(error.selectionPath)
+  const outputType = error.outputType
+
+  const object = argsTree.arguments.getDeepSelectionParent(selectionPath)?.value
+  if (object) {
+    object.getField(field)?.markAsError()
+
+    if (outputType) {
+      for (const field of outputType.fields) {
+        if (field.isRelation) {
+          object.addSuggestion(new ObjectFieldSuggestion(field.name, 'true'))
+        }
+      }
+    }
+  }
+
+  argsTree.addErrorMessage((chalk) => {
+    let msg = `Invalid scalar field ${chalk.redBright(`\`${field}\``)} for ${chalk.bold('include')} statement`
+    if (outputType) {
+      msg += ` on model ${chalk.bold(outputType.name)}. ${availableOptionsMessage(chalk)}`
+    } else {
+      msg += '.'
+    }
+
+    msg += `\nNote, that ${chalk.bold('include')} statements only accept relation fields.`
+    return msg
+  })
+}
+
+function applyEmptySelectionError(error: EmptySelectionError, argsTree: ArgumentsRenderingTree) {
+  const outputType = error.outputType
+  const selection = argsTree.arguments.getDeepSelectionParent(error.selectionPath)?.value
+  const isEmpty = selection?.isEmpty() ?? false
+
+  if (selection) {
+    selection.removeAllFields()
```

Comment:
Why must the fields be removed here?

Reply:
So, this a case where we have fields in `select`, but all of them are `false`, so it results in an empty selection which is still not valid. In that case, we have several options on how to render suggestions:

1. Display only those fields that do not already exist on selection. Example rendering:
   
   ```ts
   {
     select: {
        id: false
        posts: false,
    ?   name?: true 
    ?   email?: true 
     }
   }
   ```

   There are couple of possible problems here. First, we are assuming that user needs to add new field to the selection, rather than change one of the existing ones to `true`. Second, we might end up in a situation where all fields are already used in selection and we have nothing left to suggest.

2. Completely ignore users input and suggest all the fields. Example rendering:

    ```ts
   {
     select: {
    ?   id?: true
    ?   posts?: true,
    ?   name?: true 
    ?   email?: true 
     }
   }
   ```

   In that case, we are suggesting that user can either change one of the existing fields to `true` or add a new one that would be `true`. And this is the approach I've taken here. In order to do it, we first remove all the fields user specified here and then add them (+ other fields from suggestion) as suggestion.

Reply:
Nice, thanks for the thorough explanation.

Reply:
I would have loved that in the comments.

Reply:
Added.